### ISSUE TITLE
antidote update no longer works when run non-interactively 

### ISSUE THREAD

**Issue by getchoo** [mattmc3/antidote#57]
as of 1.6.3, antidote will no longer update bundles when running with `zsh -c`

example:
```sh
$ zsh -c 'source $ZDOTDIR/.antidote/antidote.zsh && antidote update'
Updating bundles...
Updating antidote...
antidote version 1.6.3 (679502a)
```

a bit of a nitpick, but i like using antidote this way to update it easily through non-zsh (bash) scripts

**Comment by mattmc3** [comment]
That’s not a nitpick. It seems perfectly reasonable that antidote should work in non-interactive mode. I’ll dig in soon and let you know what I find. Assuming there’s a reasonable fix, I’ll add a test to protect this functionality from breaking going forward. Thanks for the bug report.

**Comment by mattmc3** [comment]
## TLDR;

I think you customized antidote, and running `zsh -c` doesn't pick those customizations up. Prior versions of antidote used aggressive globbing (`**`) that handled both 'friendly names' and the default names. I suspect running `zsh -ci 'source $ZDOTDIR/.antidote/antidote.zsh && antidote update'` will fix your issue in the short term. Please confirm.

## The longer version

I was AFK for most of yesterday, but finally got around to looking into this this AM. At first I was not able to reproduce the issue, which leads me to believe that antidote is working as expected. As I dug deeper, I think I can guess what's going on. I'm going to bet that you have customized antidote's config by:

- choosing a new `ANTIDOTE_HOME`
- set the `zstyle ':antidote:bundle' use-friendly-names 'yes'`

When you run `zsh -c`, your .zshrc isn't run, meaning that antidote is using its defaults instead of your customizations. If you export your `ANTIDOTE_HOME` variable, your subshell will pick that up. Unfortunately, I don't think you can easily export your `zstyles` the same way. So, you have a couple options.

The first is to make sure your zsh command configures antidote the same way:

```zsh
$ export ANTIDOTE_HOME=/path/to/your/new/home
$ zsh -c 'source $ZDOTDIR/.antidote/antidote.zsh; zstyle ":antidote:bundle" use-friendly-names yes; antidote update'
```

The second option is to run `zsh -ci` (`i` for interactive) which will pick up your .zshrc config:

```zsh
zsh -ci 'source $ZDOTDIR/.antidote/antidote.zsh && antidote update'
```

You were able to get away with this in prior versions of antidote because I just used aggressive globbing for updates. I switched from that to make sure submodules didn't get updated. I'm debating moving back to aggressive globbing and checking for submodules, but for now `zsh -ci` is the recommendation. Let me know if this make sense.

**Comment by getchoo** [comment]
both solutions worked! while i didn't have `ANTIDOTE_HOME` set, i did have `use-friendly-names` and adding that to a `-c` command or just using `-ci` replicated the same behavior as previous versions. i can also see why excessive globbing was removed, and imo that's a great solution and it should probably stay that way.

and just a sidenote: i'm not sure if  `zsh -ci` would be the best recommendation here since at worse, it could have unintended side effects depending on what is in .zshrc (maybe something like p10k's instant prompt being run, running neofetch, etc) that would detract from just updating the bundles; at the very least though, it would most likely cause slowdowns since it would be sourcing all of your plugins. i found putting the zstyle line in .zshenv (since that will always be sourced) worked the best as you wouldn't need to use `zsh -ci` or write the line in the `zsh -c` arguments

thanks for the help!
